/* -------------------------------------------------------------------------- */
/* Copyright 2002-2025, OpenNebula Project, OpenNebula Systems                */
/*                                                                            */
/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
/* not use this file except in compliance with the License. You may obtain    */
/* a copy of the License at                                                   */
/*                                                                            */
/* http://www.apache.org/licenses/LICENSE-2.0                                 */
/*                                                                            */
/* Unless required by applicable law or agreed to in writing, software        */
/* distributed under the License is distributed on an "AS IS" BASIS,          */
/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
/* See the License for the specific language governing permissions and        */
/* limitations under the License.                                             */
/* -------------------------------------------------------------------------- */
syntax = "proto3";

import "shared.proto";

package one.market;

option go_package = "github.com/OpenNebula/one/src/oca/go/src/goca/api/marketplace";

message AllocateRequest
{
  string session_id = 1;
  string template   = 2;
}

message DeleteRequest
{
  string session_id = 1;
  int32 oid         = 2;
}

message InfoRequest
{
  string session_id = 1;
  int32 oid         = 2;
  bool decrypt      = 3;
}

message UpdateRequest
{
  string session_id = 1;
  int32 oid         = 2;
  string template   = 3;
  int32 append      = 4;
}

message RenameRequest
{
  string session_id = 1;
  int32 oid         = 2;
  string name       = 3;
}

message ChmodRequest
{
  string session_id   = 1;
  int32 oid           = 2;
  sint32 user_use     = 3;
  sint32 user_manage  = 4;
  sint32 user_admin   = 5;
  sint32 group_use    = 6;
  sint32 group_manage = 7;
  sint32 group_admin  = 8;
  sint32 other_use    = 9;
  sint32 other_manage = 10;
  sint32 other_admin  = 11;
}

message ChownRequest
{
  string session_id = 1;
  int32 oid         = 2;
  sint32 user_id    = 3;
  sint32 group_id   = 4;
}

message EnableRequest
{
  string session_id = 1;
  int32 oid         = 2;
  bool enable       = 3;
}

message AllocateDBRequest
{
  string session_id = 1;
  string xml        = 2;
}

message UpdateDBRequest
{
  string session_id = 1;
  int32 oid         = 2;
  string xml        = 3;
}

message PoolInfoRequest
{
  string session_id = 1;
}

service MarketPlaceService {
  rpc Allocate (one.market.AllocateRequest) returns (one.ResponseID);

  rpc Delete (one.market.DeleteRequest) returns (one.ResponseID);

  rpc Info (one.market.InfoRequest) returns (one.ResponseXML);

  rpc Update (one.market.UpdateRequest) returns (one.ResponseID);

  rpc Rename (one.market.RenameRequest) returns (one.ResponseID);

  rpc Chmod (one.market.ChmodRequest) returns (one.ResponseID);

  rpc Chown (one.market.ChownRequest) returns (one.ResponseID);

  rpc Enable (one.market.EnableRequest) returns (one.ResponseID);

  rpc AllocateDB (one.market.AllocateDBRequest) returns (one.ResponseID);

  rpc UpdateDB (one.market.UpdateDBRequest) returns (one.ResponseID);

  rpc PoolInfo (one.market.PoolInfoRequest) returns (one.ResponseXML);
}
